### PR TITLE
docs: typescript login

### PR DESCRIPTION
**Pull request by peintnermax**
This PR improves the documentation about the new login UI, incl. explaination of the architecture.

### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 1]
```diff
@@ -5,28 +5,67 @@ sidebar_label: Typescript Repository
...
 - `@zitadel/react`: shared React utilities and components built with Tailwind CSS
 - `@zitadel/next`: shared Next.js utilities
 - `@zitadel/tsconfig`: shared `tsconfig.json`s used throughout the monorepo
 - `eslint-config-zitadel`: ESLint preset
 
-## Implemented features of OIDC Standard
+## Login UI
+
+The new implementation of the login is based on Next.js and implements the [OIDC standard](./oidc-standard) by proxying requests to the ZITADEL backend and using the session API.
+The UI shows branding and behaves based on the policies and settings your instance or organization (when using an org scope) has.
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:22]:
```suggestion
The UI uses the branding and behaves based on the policies and settings of your instance or organization.
```

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 2]
```diff
@@ -5,28 +5,67 @@ sidebar_label: Typescript Repository
...
+It is also able to autenticate users without an OIDC context by navigating to the host of the login, to allow them to self service their account.
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:23]:
```suggestion
It is also allows users to authenticate directly (without an OIDC context) and self service their account.
```

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 3]
```diff
@@ -5,28 +5,67 @@ sidebar_label: Typescript Repository
...
+It's set of features will increase with time and its progress will be tracked [here](./typescript-repo#implemented-features).
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:24]:
```suggestion
Its set of features will increase with time and its progress will be tracked [here](./typescript-repo#implemented-features).
```

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 4]
```diff
@@ -5,28 +5,67 @@ sidebar_label: Typescript Repository
...
+
+### Architectural Overview
+
+The new login app consists of a confidential (backend) part and a browser (client side) part.
+In addition to that, the OIDC specification builds upon a proxy which is set up in the Next.js middleware. It rewrites requests of the paths `/.well-known/:path*`, `/oauth/:path*`, `/oidc/:path*` to ZITADEL.
+
+The backend side of the application is authorized using a machine user and a personal access token.
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:31]:
```suggestion
The backend side of the application is authorized using a machine user's personal access token.
```

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 5]
```diff
@@ -5,28 +5,67 @@ sidebar_label: Typescript Repository
...
+
+_Note that the illustration is just a representation of the architecture and may not specify actual endpoints._
+
+The flow starts by the `/authorize` endpoint which is intercepted by the login and forwarded to ZITADEL.
+ZITADEL interprets the request and specifically redirects back to the login app.
+It does so by redirecting to `/login`.
+The login is then able to load an [AuthRequest](/docs/apis/resources/oidc_service/oidc-service-get-auth-request#get-oidc-auth-request-details).
+
+The Auth Request defines how users proceed to authenticate. If no special prompts or scopes are set, the login brings up the `/loginname` page.
+The /loginname page allows to enter loginname or email of a user. User discovery is implemented at /api/loginname and if the user is found, redirected to the available authentication method page.
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:49]:
```suggestion
The /loginname page allows to enter loginname or email of a user. User discovery is implemented at /api/loginname and if the user is found, they will be redirected to the available authentication method page.
```

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 6]
```diff
@@ -5,28 +5,67 @@ sidebar_label: Typescript Repository
...
+Right after the user is found, a session is created and set as cookie.
+This cookie is then hydrated with more information once the users continues.
+The OIDC Auth request is always passed in the url to have a context to the ongoing authentication flow.
+If enough user information is retrieved and the user is authenticated, the flow is finalized by [requesting a the callback url](/docs/apis/resources/oidc_service/oidc-service-create-callback) for the auth request and the user is redirected back to the application.
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:53]:
```suggestion
If enough user information is retrieved and the user is authenticated accorind to the policies, the flow is finalized by [requesting a the callback url](/docs/apis/resources/oidc_service/oidc-service-create-callback) for the auth request and the user is redirected back to the application.
```

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 7]
```diff
@@ -53,13 +92,39 @@ The repo implements the [OIDC standard](./oidc-standard)
...
 - Register
+
   - [x] Email Password
   - [x] Passkey
 
-## How to setup domains
+#### Self service
+
+The self service part of the application allows to load the login without an OIDC flow.
+After authenticating, or with a valid session, users are able to self service their account.
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:102]:
```suggestion
Authenticated users are directly able to self service their account.
```

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx [hunk 8]
```diff
@@ -53,13 +92,39 @@ The repo implements the [OIDC standard](./oidc-standard)
...
 After your domain has been verified, you can reconfigure your DNS settings in order to deploy the login on your own.
+
+#### Deploy to Vercel
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2Fzitadel%2Ftypescript&env=ZITADEL_API_URL,ZITADEL_SERVICE_USER_ID,ZITADEL_SERVICE_USER_TOKEN&root-directory=apps/login&envDescription=Setup%20a%20service%20account%20with%20IAM_OWNER%20membership%20on%20your%20instance%20and%20provide%20its%20id%20and%20personal%20access%20token.&project-name=zitadel-login&repository-name=zitadel-login)
+
+To deploy your own version on Vercel, navigate to your instance and create a service user.
+Copy its id from the overview and set it as `ZITADEL_SERVICE_USER_ID`.
+Then create a personal access token (PAT), copy and set it as `ZITADEL_SERVICE_USER_TOKEN`, then navigate to your instance settings and make sure it gets `IAM_OWNER` permissions.
+Finally set your instance url as `ZITADEL_API_URL`. Make sure to set it without trailing slash.
```
Comment (livio-a) [inline docs/docs/guides/integrate/login-ui/typescript-repo.mdx:130]: we might need to add some images

**Review by livio-a — CHANGES_REQUESTED** [review]

File: console/src/app/modules/accounts-card/accounts-card.component.ts [hunk 9]
```diff
@@ -63,6 +63,7 @@ export class AccountsCardComponent implements OnInit {
       customQueryParams: {
         prompt: 'login',
       } as any,
+      scope: 'openid profile email urn:zitadel:iam:org:id:256088845584504464',
```
[the remark this answers is not in the document]
Comment (peintnermax) [reply console/src/app/modules/accounts-card/accounts-card.component.ts:66]: yes.. removed 👍

File: docs/docs/guides/integrate/login-ui/typescript-repo.mdx (hunk 8 above)
Comment (peintnermax) [reply docs/docs/guides/integrate/login-ui/typescript-repo.mdx:130]: I've added a screenshot

**Review by livio-a — APPROVED** [review]

**Pull request merged**